### PR TITLE
add support for 1.19

### PR DESCRIPTION
**Pull request by aclevername**
### Description

This PR is for release-0.38
https://github.com/weaveworks/eksctl/pull/3061 previous PR for main

### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [ ] Manually tested
- [ ] Made sure the title of the PR is a good description that can go into the release notes
- [ ] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: integration/tests/addons/addons_test.go [hunk 1]
```diff
@@ -36,8 +36,8 @@ var _ = Describe("(Integration) [EKS Addons test]", func() {
 		BeforeEach(func() {
 			clusterConfig := api.NewClusterConfig()
 			clusterConfig.Metadata.Name = clusterName
-			clusterConfig.Metadata.Version = "1.18"
-			clusterConfig.Metadata.Region = api.DefaultRegion
+			clusterConfig.Metadata.Version = "1.19"
+			clusterConfig.Metadata.Region = params.Region
```
Comment (aclevername) [inline integration/tests/addons/addons_test.go:40]: bug I spotted while running the integration tests

File: pkg/actions/addon/get.go [hunk 2]
```diff
@@ -99,6 +99,10 @@ func (a *Manager) findNewerVersions(addon *api.Addon) (string, error) {
 		return "", err
 	}
 
+	if len(versions.Addons) == 0 {
+		return "-", nil
+	}
+
```
Comment (aclevername) [inline pkg/actions/addon/get.go:105]: while running integration tests it was the first time we've not had a new version of vpc-cni available, so only just spotted this panic that needed fixing.

**Review by Callisto13 — APPROVED** [review]

**Comment by aclevername** [comment]
just spotted I forgot to commit the `coredns-1.19` yaml asset, going to add that now.

**Review by michaelbeaumont — APPROVED** [review]

**Review by michaelbeaumont — APPROVED** [review]

File: pkg/addons/default/assets/coredns-1.19.json [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+                ]
+              }
+            },
+            "containers": [
+              {
+                "args": [
+                  "-conf",
+                  "/etc/coredns/Corefile"
+                ],
+                "image": "%s.dkr.ecr.%s.%s/eks/coredns:v1.7.0-eksbuild.1",
```
Comment (bkempe) [inline pkg/addons/default/assets/coredns-1.19.json:167]: according to https://docs.aws.amazon.com/eks/latest/userguide/update-cluster.html `1.8.0` is recommended here

Comment (aclevername) [reply pkg/addons/default/assets/coredns-1.19.json:167]: Thanks! Will update this.

**Pull request merged**